sycl/test-e2e/Graph/RecordReplay/in_order_queue_with_host_managed_dependencies_memcpy.cpp

```diff
@@ -0,0 +1,66 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// Tests submitting memcpy to an in-order queue before recording
+// commands from it.
+
+#include "../graph_common.hpp"
+
+int main() {
+  using T = int;
+
+  queue Queue{{sycl::ext::intel::property::queue::no_immediate_command_list{},
+               sycl::property::queue::in_order{}}};
+
+  if (!are_graphs_supported(Queue)) {
+    return 0;
+  }
+
+  // Check if device has usm shared allocation
+  if (!Queue.get_device().has(sycl::aspect::usm_shared_allocations))
+    return 0;
+
+  std::vector<T> TestDataIn(Size);
+  T *TestData = sycl::malloc_shared<T>(Size, Queue);
+  T *TestDataOut = sycl::malloc_shared<T>(Size, Queue);
+
+  ext::oneapi::experimental::command_graph Graph{Queue.get_context(),
+                                                 Queue.get_device()};
+
+  std::memset(TestDataIn.data(), 1, Size * sizeof(T));
+  Queue.memcpy(TestData, TestDataIn.data(), Size * sizeof(T));
+
+  Graph.begin_recording(Queue);
+
+  auto GraphEvent = Queue.submit([&](handler &CGH) {
+    CGH.single_task<class TestKernel2>([=]() {
+      for (size_t i = 0; i < Size; i++) {
+        TestData[i] += static_cast<T>(i);
+      }
+    });
+  });
+
+  Graph.end_recording(Queue);
+
+  auto GraphExec = Graph.finalize();
+  Queue.submit([&](handler &CGH) { CGH.ext_oneapi_graph(GraphExec); });
+
+  Queue.memcpy(TestDataOut, TestData, Size * sizeof(T));
+
+  Queue.wait_and_throw();
+
+  std::vector<T> Reference(Size);
+  std::memset(Reference.data(), 1, Size * sizeof(T));
+  for (size_t i = 0; i < Size; i++) {
+    Reference[i] += i;
```

Comment:
You are right. Thanks for the suggestion. `Reference` has been removed.

Reply:
I was too hasty in deleting `Reference`.
In fact, References were not equal to 1 but to 0x1010101 because the memset works with bytes.
I prefer to keep memset than a hard value, in case we change the size of variables in the future.
I thus updated the code to use a single `Reference` instead of an array.